core/config/config.go

```diff
@@ -38,31 +44,47 @@ var (
 	Production                      bool
 )
 
-// Config encapsulates Core-level, persistent configuration options.
-type Config struct {
-	ID                   string  `json:"id"`
-	IsSigner             bool    `json:"is_signer"`
-	IsGenerator          bool    `json:"is_generator"`
-	BlockchainID         bc.Hash `json:"blockchain_id"`
-	GeneratorURL         string  `json:"generator_url"`
-	GeneratorAccessToken string  `json:"generator_access_token"`
-	BlockHSMURL          string  `json:"block_hsm_url"`
-	BlockHSMAccessToken  string  `json:"block_hsm_access_token"`
-	ConfiguredAt         time.Time
-	BlockPub             string        `json:"block_pub"`
-	Signers              []BlockSigner `json:"block_signer_urls"`
-	Quorum               int
-	MaxIssuanceWindow    chainjson.Duration
-}
+// Load loads the stored configuration, if any, from the database.
+// It will first try to load the config from raft storage; if that fails,
+// it will try Postgres next. If it finds a config in Postgres but not in raft
+// storage, the config will be added to raft storage.
+func Load(ctx context.Context, db pg.DB, rDB *raft.Service) (*Config, error) {
+	data, err := rDB.Get(ctx, "/core/config")
+	if err != nil {
+		return nil, errors.Wrap(err)
+	}
+	if data == nil {
+		// Check Postgres next.
+		config, err := loadFromPG(ctx, db)
+		if err != nil {
+			return nil, errors.Wrap(err)
+		}
 
-type BlockSigner struct {
-	AccessToken string             `json:"access_token"`
-	Pubkey      chainjson.HexBytes `json:"pubkey"`
-	URL         string             `json:"url"`
+		// If we were able to find a config in Postgres, store it in Raft
+		if config != nil {
+			val, err := proto.Marshal(config)
+			if err != nil {
+				return nil, errors.Wrap(err)
+			}
+			err = rDB.Insert(ctx, "/core/config", val)
+			if err != nil {
+				return nil, errors.Wrap(err)
+			}
+			return config, nil
```

Comment:
Should this migration logic also delete the old config to avoid confusion? Maybe it should even drop the table???